cli/flox/src/commands/init.rs

```diff
@@ -0,0 +1,353 @@
+
+        Ok(Self::combine_customizations(customizations))
+    }
+
+    /// Deduplicate packages and concatenate hooks into a single string
+    fn combine_customizations(customizations: Vec<InitCustomization>) -> InitCustomization {
+        let mut hooks: Vec<String> = vec![];
+        // Deduplicate packages with a set
+        let mut packages_set = HashSet::<PackageToInstall>::new();
+        for customization in customizations {
+            if let Some(packages) = customization.packages {
+                packages_set.extend(packages)
+            }
+            if let Some(hook) = customization.hook {
+                hooks.push(hook)
+            }
+        }
+
+        let hook = (!hooks.is_empty()).then(|| {
+            formatdoc! {"
+                # Autogenerated by flox
+
+                {}
+
+                # End autogenerated by flox", hooks.join("\n\n")}
+        });
+
+        let packages = (!packages_set.is_empty())
+            .then(|| packages_set.into_iter().collect::<Vec<PackageToInstall>>());
+
+        InitCustomization { hook, packages }
+    }
+}
+
+trait InitHook {
+    fn should_run(&self, path: &Path) -> bool;
+
+    fn prompt_user(&self) -> Result<bool>;
+
+    fn get_init_customization(&self) -> InitCustomization;
+}
+
+struct Requirements;
+
+impl InitHook for Requirements {
+    fn should_run(&self, path: &Path) -> bool {
+        path.join("requirements.txt").exists()
+    }
+
+    fn prompt_user(&self) -> Result<bool> {
+        let message = formatdoc! {"
+            Flox detected a requirements.txt
+
+            Python projects typically need:
+            * python, pip
+            * A Python virtual environment to install dependencies into
+
+            Would you like Flox to apply the standard Python environment?
+            You can always revisit the environment's declaration with 'flox edit'
+        "};
+
+        let help = "Use '--auto-setup' to apply Flox recommendations in the future.";
+
+        let dialog = Dialog {
+            message: &message,
+            help_message: Some(help),
+            typed: Select {
+                options: ["Yes (python 3.11)", "No", "Show suggested modifications"].to_vec(),
+            },
+        };
+
+        let (mut choice, _) = dialog.raw_prompt()?;
+
+        if choice == 2 {
+            let message = formatdoc! {"
+
+            {}
+            Would you like Flox to apply these modifications?
+            You can always revisit the environment's declaration with 'flox edit'
+        ", format_customization(&self.get_init_customization())?};
+
+            let help = "Use '--auto-setup' to apply Flox recommendations in the future.";
+
+            let dialog = Dialog {
+                message: &message,
+                help_message: Some(help),
+                typed: Select {
+                    options: ["Yes (python 3.11)", "No"].to_vec(),
+                },
+            };
+
+            (choice, _) = dialog.raw_prompt()?;
+        }
```

Comment:
nit: id personally split these prompts and messages a bit

```suggestion
        let message = formatdoc! {"
            Flox detected a requirements.txt

            Python projects typically need:
            * python, pip
            * A Python virtual environment to install dependencies into
        "};
        
        let prompt = formatdoc!{"
        	Would you like Flox to apply the standard Python environment?
            You can always revisit the environment's declaration with 'flox edit'
		"};
		
        let help = "Use '--auto-setup' to apply Flox recommendations in the future.";

		message::plain(message);

        let dialog = Dialog {
            message: &prompt,
            help_message: Some(help),
            typed: Select {
                options: ["Yes (python 3.11)", "No", "Show suggested modifications"].to_vec(),
            },
        };

        let (mut choice, _) = dialog.raw_prompt()?;

        if choice == 2 {
            let message = format_customization(&self.get_init_customization())?;
            message::plain(message);
            
            let prompt = formatdoc! {"
           	    Would you like Flox to apply these modifications?
                You can always revisit the environment's declaration with 'flox edit'
        	"};

            let help = "Use '--auto-setup' to apply Flox recommendations in the future.";

            let dialog = Dialog {
                message: &prompt,
                help_message: Some(help),
                typed: Select {
                    options: ["Yes (python 3.11)", "No"].to_vec(),
                },
            };

            (choice, _) = dialog.raw_prompt()?;
        }
```

Reply:
I think the whole thing should be perceived as one message. `Flox detected a requirements.txt` all the way to `Would you like Flox to apply the standard Python environment?` If we split them up, they currently get different formatting, so the beginning of the message is more likely to be ignored.